java/com/jfreechartextensions/renderers/ExtendedXYStackedAreaRenderer.java

```diff
@@ -23,156 +24,223 @@
-                    Stack tip1 = areaState.getLastSeriesPoints();
+                double transY2 = rangeAxis.valueToJava2D(ph1, dataArea,
+                        plot.getRangeAxisEdge());
 
-                    while(!tip1.empty()) {
-                        Point generator = (Point)tip1.pop();
-                        areaState.getSeriesArea().addPoint((int)generator.getX(), (int)generator.getY());
+                if (orientation == PlotOrientation.VERTICAL) {
+                    // Add the last point (x,0)
+                    areaState.getSeriesArea().addPoint((int) transX1,
+                            (int) transY2);
+                }
+                else if (orientation == PlotOrientation.HORIZONTAL) {
+                    // Add the last point (x,0)
+                    areaState.getSeriesArea().addPoint((int) transY2,
+                            (int) transX1);
+                }
+
+                // Add points from last series to complete the base of the
+                // polygon
+                if (series != 0) {
+                    Stack points = areaState.getLastSeriesPoints();
+                    while (!points.empty()) {
+                        Point point = (Point) points.pop();
+                        areaState.getSeriesArea().addPoint((int) point.getX(),
+                                (int) point.getY());
                     }
                 }
 
+                //  Fill the polygon
                 g2.setPaint(seriesFillPaint);
                 g2.setStroke(seriesStroke);
                 g2.fill(areaState.getSeriesArea());
-                if(this.isOutline()) {
-                    g2.setStroke(this.lookupSeriesOutlineStroke(series));
-                    g2.setPaint(this.lookupSeriesOutlinePaint(series));
+
+                //  Draw an outline around the Area.
+                if (isOutline()) {
+                    g2.setStroke(lookupSeriesOutlineStroke(series));
+                    g2.setPaint(lookupSeriesOutlinePaint(series));
                     g2.draw(areaState.getSeriesArea());
                 }
             }
 
-            int shape2 = plot.getDomainAxisIndex(domainAxis);
-            int entities = plot.getRangeAxisIndex(rangeAxis);
-            this.updateCrosshairValues(crosshairState, x1, ph1 + y1, shape2, entities, transX1, transY1, orientation);
-        } else if(pass == 1) {
-            Object shape3 = null;
-            //if(this.getPlotShapes()) {
-            if(this.getPlotShapes(series)) {
-                shape3 = this.getItemShape(series, item);
-                if(plot.getOrientation() == PlotOrientation.VERTICAL) {
-                    shape3 = ShapeUtilities.createTranslatedShape((Shape)shape3, transX1, transY1);
-                } else if(plot.getOrientation() == PlotOrientation.HORIZONTAL) {
-                    shape3 = ShapeUtilities.createTranslatedShape((Shape)shape3, transY1, transX1);
-                }
+            int datasetIndex = plot.indexOf(dataset);
```

Comment:
Do we want to have state.getDataset() instead of dataset.
I don't have any idea, just want to draw your attention.